File: src/main/java/com/thiagoh/cci/c1/Q1_3.java
package com.thiagoh.cci.c1;

public class Q1_3 {

	public static char[] removeDuplicates(char[] a) {
		if (a == null || a.length <= 1) {
			return a;
		}
		int p = 1;
		int n = 1;
		while (n < a.length) {
			boolean b = false;
			for (int i = 0; i < p && !b; i++)
				if (a[n] == a[i])
					b = true;
			if (!b) {
				a[p++] = a[n];
			}
			n++;
		}
		while (p < a.length)
			a[p++] = '\0';
		return a;
	}

	public static char[] removeDuplicates2(char[] s) {
		if (s == null || s.length <= 1) {
			return s;
		}
		return s;
	}
}

File: src/test/python/com/thiagoh/cci/c1/Q1_3.py
__author__ = 'thiago'

import unittest

from src.main.python.com.thiagoh.cci.c1.Q1_3 import *


class MyTestCase(unittest.TestCase):
    def test_something(self):
        self.assertEqual(removeDuplicates(""), "");
        self.assertEqual(removeDuplicates(None), None);
        self.assertEqual(removeDuplicates("a"), "a");
        self.assertEqual(removeDuplicates("ab"), "ab");
        self.assertEqual(removeDuplicates("aab"), "ab\0");
        self.assertEqual(removeDuplicates("aaab"), "ab\0\0");

        s = "qwertyuiopasdfghjklzxcvbnm";
        sb = list();
        for i in range(0, len(s)): sb.append('\0');

        self.assertEqual(removeDuplicates(s), s);
        self.assertEqual(removeDuplicates(s + s), s + (''.join(sb)));

if __name__ == '__main__':
    unittest.main()
